### DISCUSSION TITLE
how to build a nuxt 3 project that uses hybrid rendering

### DISCUSSION THREAD

**Discussion by kasunAththanagoda** [nuxt/nuxt#25001] [Questions]
i am a beginner to nuxt and currently trying to migrate a vue js 3 project to nuxt js 3 .i have gone through many documents and trying to use hybrid rendering to my project . Can please anyone explain or suggest a source for me about how can i build the project or what are the things i should consider.

**Comment by Mobbbb** [comment]
1、change `vue-router` to [custom-routing](https://nuxt.com/docs/guide/going-further/custom-routing#router-options)
2、change `fetch request` to `useFetch`
3、note the parameters that need to be initialized
4、don't use document, window etc.. when `useFetch`

**Reply by TheAlexLichter** [reply]
(There is no need to change to custom routing)

**Reply by Mobbbb** [reply]
change the following code
```javascript
import { createRouter, createWebHashHistory } from 'vue-router'

const router = createRouter({
    history: createWebHashHistory(),
    routes,
})
export default router
```
to
```typescript
import type { RouterConfig } from '@nuxt/schema'

export default <RouterConfig> {
    routes,
}
```
is not required?

**Reply by TheAlexLichter** [reply]
No, why would it?
This actually hurts SEO 🙈

**Reply by Mobbbb** [reply]
Is using file-system routing the best practice?

**Reply by TheAlexLichter** [reply]
That's why it is the default. If it does not match your needs you can always manually set up your routes though, but `WebHashHistory` will hurt SEO.

**Reply by Mobbbb** [reply]
well, thanks! But for migrating a vue js 3 project to nuxt js 3, it looks like the workload will be very heavy

**Reply by TheAlexLichter** [reply]
@Mobbbb Depends on the current structure. You don't *have* to migrate to file-based routing but it can pay off in terms of DX and more. Especially if your routes are already created with that structure in mind.

**Comment by Mobbbb** [comment]
By the way, you might also need to change how components are referenced.
